pkg/ddevapp/traefik_config_template.yaml

```diff
@@ -2,61 +2,63 @@
 
 http:
   routers:
-    {{$appname := .App.Name}}{{ range $s := .RoutingTable }}
-    {{ if not $s.HTTPS }}
-    {{$appname}}-{{$s.InternalServiceName}}-{{$s.InternalServicePort}}-http:
-      {{if not $.UseLetsEncrypt}}{{/* Let's Encrypt only works with Host(), but we need HostRegex() for wildcards*/}}
+    {{ $appname := .App.Name}}{{ range $s := .RoutingTable }}
+    {{- if not $s.HTTPS -}}
+    {{ $appname }}-{{ $s.Service.InternalServiceName }}-{{ $s.Service.InternalServicePort }}-http:
+      entrypoints:
+        - http-{{$s.ExternalPort}}
+      {{- if not $.UseLetsEncrypt -}}{{/* Let's Encrypt only works with Host(), but we need HostRegexp() for wildcards*/}}
       rule: HostRegexp({{ range $i, $h := $s.ExternalHostnames }}{{if $i}}, {{end}}`{{$h}}`{{end}})
```

Comment:
You'll need to explain later why that's important, include it in a separate PR. I don't think the len() is required, because `range` does nothing if there's nothing there. 

Please, when you make a comment on something like this, explain why you think it's important. Here, you could say "this is a bug" or "doesn't work" because ...But since this has been working as is (I think) for a couple of years, it's hard for me to understand what you're saying or asking for.